test/topology/conftest.py

```diff
@@ -6,10 +6,75 @@
 # This file configures pytest for all tests in this directory, and also
 # defines common test fixtures for all of them to use
 
+import asyncio
 import pathlib
 import sys
+from typing import AsyncGenerator
+from test.pylib.random_tables import RandomTables                        # type: ignore
+import pytest
+from cassandra.cluster import Session, ResponseFuture                    # type: ignore
 
 # Add test.pylib to the search path
 sys.path.append(str(pathlib.Path(__file__).resolve().parents[1]))
 
 
+def _wrap_future(f: ResponseFuture) -> asyncio.Future:
+    """Wrap a cassandra Future into an asyncio.Future object.
+
+    Args:
+        f: future to wrap
+
+    Returns:
+        And asyncio.Future object which can be awaited.
+    """
+    loop = asyncio.get_event_loop()
+    aio_future = loop.create_future()
+
+    def on_result(result):
+        loop.call_soon_threadsafe(aio_future.set_result, result)
+
+    def on_error(exception, *_):
+        loop.call_soon_threadsafe(aio_future.set_exception, exception)
+
+    f.add_callback(on_result)
+    f.add_errback(on_error)
+    return aio_future
+
+
+def run_async(self, *args, **kwargs) -> asyncio.Future:
+    return _wrap_future(self.execute_async(*args, **kwargs))
+
+
+Session.run_async = run_async
```

Comment:
> This monkey-patching applies only to tests which explicitly import test/topology/conftest.py, right?
...
> Unless run_async is not an existing member in Session, we're adding a completely new member? Then it doesn't matter.

Yes, nothing else is using it.

> Doesn't pytest use this file internally somehow? I'm worried if pytest won't accidentally apply this patch to other test modules as well due to some pytest magic.

No, this is the cassandra driver. And pytest processes are executed independently for each pytest file. There is no way for this to affect anything outside topology test directory. It was before but no longer. This series reverts previous async changes to top level conftest wich _could_ affect other tests but it _did not_.

> If it would, maybe a way to protect ourselves from that is to apply the patch in the cql fixture.

If monkey patching is not wanted (even though it's used elsewhere), I would rather subclass `cassandra.Session`. Let me know.

Reply:
> If monkey patching is not wanted (even though it's used elsewhere), I would rather subclass cassandra.Session than keep adding fixture hacks. Let me know.

No, I don't think doing more work here would be valuable, let's leave it as it is.